### PR TITLE
Toggle signed-in state in the top nav

### PR DESCRIPTION
**Pull request by cnunciato**
Add a `pulumi-user-toggle` component that lets you toggle content based on whether the user is currently signed in. Tracks with current cookie state (so when users sign out in another tab, all tabs are updated). Also adds the component to the top nav to toggle "Sign In" / "Pulumi Console" accordingly. (See [this thread](https://pulumi.slack.com/archives/C01E2KK808G/p1681484915209379) for related discussion on the labeling choice.)

https://user-images.githubusercontent.com/274700/232101395-f566f7e2-b509-4d7a-8ef6-ff0e35a4b60f.mov

## Checklist:

- [x] I have reviewed the [style guide](../STYLE-GUIDE.md).
- [x] If blogging, I have reviewed the [blogging guide](../BLOGGING.md).
- [x] I have manually confirmed that all new links work.
- [x] I added aliases (i.e., redirects) for all filename changes.

### CODE REVIEW

File: themes/default/layouts/partials/header.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{ if hasPrefix .RelPermalink "/docs/" }}
-    {{ partial "tertiary-nav" . }}
+    {{ partial "top-nav" . }}
```
Comment (cnunciato) [inline themes/default/layouts/partials/header.html:2]: I named this wrong to begin with. (There is another thing called "tertiary nav".) 🤦 Should be right now.

File: themes/default/theme/stencil/src/store/reducers/user.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+function getUserInfoCookie(): UserInfoCookie {
+    const cookie = parseCookie();
+    const pulumiWebUserInfo = cookie["pulumi_web_user_info"];
+    let userId: string;
+
+    if (pulumiWebUserInfo) {
+        try {
+            userId = JSON.parse(decodeURIComponent(pulumiWebUserInfo).replace("j:", "")).userId;
+        } catch (error) {
+            console.log(`Failed to parse pulumi_web_user_info cookie '${pulumiWebUserInfo}'.`);
```
Comment (sean1588) [inline themes/default/theme/stencil/src/store/reducers/user.ts:24]: should this be a console.error instead?

File: themes/default/theme/stencil/src/store/actions/user.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { TypeKeys } from "./index";
+
+export interface GetUser {
+    type: TypeKeys.GET_USER_INFO;
+}
+
+export const getUser = () => (dispatch, _getState) => {
+    console.log("getUser");
```
Comment (sean1588) [inline themes/default/theme/stencil/src/store/actions/user.ts:8]: im guessing this was this in there just for testing?

Comment (cnunciato) [reply themes/default/theme/stencil/src/store/actions/user.ts:8]: Ah good catch!

File: themes/default/theme/src/user.ts
Comment (cnunciato) [inline themes/default/theme/src/user.ts:1]: Gotta remove this also.

**Review by zchase — APPROVED** [review]
LGTM! Can always follow up once the username is available. Thanks for doing this!

**Pull request merged**